src/newTest/java/org/opensearch/test/AbstractIntegrationTest.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.test;
+
+import org.junit.runner.RunWith;
+import org.opensearch.test.framework.TestSecurityConfig;
+import org.opensearch.test.framework.TestSecurityConfig.Role;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class AbstractIntegrationTest {
+
+	/**
+	 * Auth domain with HTTPS Basic and the internal user backend
+	 */
+	protected final static TestSecurityConfig.AuthcDomain AUTHC_HTTPBASIC_INTERNAL = new TestSecurityConfig.AuthcDomain("basic", 0)
```

Comment:
For the AuthcDomain(...) or httpAuthenticator(...), what would it take to use enums instead of string values?

Reply:
Yes, we should introduce enums here. Also, I removed the abstract base class and moved the constants to the User and Authc classes, we should prefer composition over inheritance.